esp-rtos/src/timer/mod.rs

```diff
@@ -170,7 +173,7 @@ impl TimeDriver {
         let sleep_duration = next_wakeup.saturating_sub(now);
 
         // assume 52-bit underlying timer. it's not a big deal to sleep for a shorter time
-        let mut timeout = sleep_duration & ((1 << 52) - 1);
+        let mut timeout = sleep_duration.min((1 << 52) - 1);
```

Comment:
Maybe turning a >52-bit wakeup time into a potential 0 isn't the best thing to do :)